src/interface.ts

```diff
@@ -72,7 +104,23 @@ interface DesktopAgent {
    * Returns a promise that resolves to an Array. The resolved dataset & metadata is Desktop Agent-specific.
    * If the resolution errors, it returns an `Error` with a string from the `ResolveError` enumeration.
    */
-  resolve(intent: IntentName, context: Context): Promise<Array<AppMetadata>>;
+  resolveIntent(intent: Intent): Promise<ResolveIntentResults>;
```

Comment:
I don't understand the case of resolving only an intent. If you already know the intent name you can just send it without having to resolve anything. What use case does it cover?

For me, resolving the intents+applications for a context is the use case that makes sense.

Reply:
The intent object has a context property.  Having both a property and an argument was confusing and I couldn't figure out which one was the one to use.

This is a recurring theme that we should agree on.  I'm really OK with either approach:

Reply:
I like being explicit about the intent and the context being different things to clear up this kind of ambiguity, as I guess you have gathered by now :-)

* You send an _intent_ + an associated _context_ (but the context is not part of the intent)
* You resolve a _context_, resulting in a _list of intents_ (and the applications declaring them, potentially)

Reply:
I think we are very much on the same page :) Now we just need to get some consensus.